util/no_destructor.h

```diff
@@ -18,12 +18,7 @@ class NoDestructor {
  public:
   template <typename... ConstructorArgTypes>
   explicit NoDestructor(ConstructorArgTypes&&... constructor_args) {
-    static_assert(sizeof(instance_storage_) >= sizeof(InstanceType),
```

Comment:
I don't consider the `static_asserts` to be redundant. They check that the buffer is suitable, at the site of the `new` operator. I would like to keep those checks, given that we've run into toolchains with bugs in `alignas`.

Reply:
I'll add them back.